teach_base/lesson_02/05_day_of_the_year.rb

```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+
+# Three numbers are specified, which indicate the day, month, year
+# (we ask the user). Find the ordinal date starting from the beginning
+# of the year. Take into account that the year may be a leap year.
+# A leap year, if it is divisible by four, but if it is divided by
+# 100 without remainder, it is not a leap year. However, if it is divisible
+# by 400, it is a leap year.(It is forbidden to use built-in ruby methods
+# for this like Date # yday or Date # leap?)
+
+puts 'Please input the date(day, month, year)!'
+day = gets.chomp.to_i
+month = gets.chomp.to_i
+year = gets.chomp.to_i
+
+def uncorrect_date?(day, month, year)
```

Comment:
This method doesn't seem to work properly.
It says that 31.02.19 is a correct date (But it's not).
But 14.07.-2200 is incorrect (but it is)

Try to optimize the condition itself. It has too many ||